.github/actions/maven-build-and-tag/action.yml

```diff
@@ -117,6 +131,16 @@ runs:
       run: |
         GITHUB_PR_NUMBER=PR-${{ github.event.pull_request.number }}
         echo "preview-version=0.0.1-$GITHUB_PR_NUMBER-SNAPSHOT" >> $GITHUB_OUTPUT
+        echo "GITHUB_PR_NUMBER=$GITHUB_PR_NUMBER" >> $GITHUB_ENV
+
+    - uses: actions/cache/restore@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
```

Comment:
It would be clearer if the other cache/restore action was moved next to this one, and a comment or name added here to explain why a distinct restore is done with another key

Reply:
Added name to the step to document the intention.